core-addon/FirefoxPrivilegedApi.js

```diff
@@ -26,14 +39,41 @@ this.firefoxPrivilegedApi = class extends ExtensionAPI {
           let augmentedOptions = options;
           augmentedOptions.useEncryption = true;
 
+          // eslint-disable-next-line no-undef
           TelemetryController.submitExternalPing(type, payload, options);
         },
         async generateUUID() {
           // eslint-disable-next-line no-undef
           let str = gUUIDGenerator.generateUUID().toString();
           return str.substring(1, str.length - 1);
-        }
-      }
-    }
+        },
+        async getRemoteSettings() {
+          // eslint-disable-next-line no-undef
+          return RemoteSettings(STUDY_COLLECTION_KEY).get();
+        },
+        // eslint-disable-next-line no-undef
+        onRemoteSettingsSync: new ExtensionCommon.EventManager({
+          context,
+          name: "firefoxPrivilegedApi.onRemoteSettingsSync",
+          // Fire a callback to listener when remote settings collection updates, passing the contents of the collection.
+          register: (fire) => {
+            const callback = (current) => {
+              fire.async(current).catch(() => {}); // ignore Message Manager disconnects
+            };
+            // eslint-disable-next-line no-undef
+            RemoteSettings(STUDY_COLLECTION_KEY).on("sync", async (event) => {
+              let {
+                data: { current },
+              } = event;
+              callback(current);
+            });
+            return () => {
+              // eslint-disable-next-line no-undef
+              RemoteSettings(STUDY_COLLECTION_KEY).off("sync");
```

Comment:
I just realize that `.off()` expects a callback reference :( https://searchfox.org/mozilla-central/rev/69babd862de70cabfa1d0a369d38e4881bd41e4d/services/settings/RemoteSettingsClient.jsm#114